src/foam/core/console/FileUpload.js

```diff
@@ -0,0 +1,1244 @@
+              start('strong').add(this.uploadedFiles$.map(files => `Files uploaded: ${files ? files.length : 0}`)).end().
+              show(this.filesVerified$).
+              start('span').addClass('ml-sm').addClass(this.myClass('text-success')).
+                add(' ✅ Structure verified').
+              end().
+            end().
+          end().
+        
+        // Manual input section
+        start('div').addClass('mt-lg').
+          show(this.uploadedFiles$.map(files => !files || files.length === 0)).
+          start('h4').add('Or Manual Content Input').end().
+          start('p').addClass(this.myClass('text-muted')).
+            add('Paste your file content directly if not using file upload.').
+          end().
+          start('div').addClass('mt-lg').
+            start('label').add('Format: ').end().
+            add(this.FORMAT).
+            show(this.format$.map(f => f === 'CSV')).
+            start('label').addClass('ml-lg').add('Delimiter: ').end().
+            add(this.DELIMITER).
+          end().
+          start('div').addClass('mt-lg').
+            add(this.INPUT).
+          end().
+        end().
+      end().
+      
+      // Structure Analysis & DAO Selection
+      start('div').addClass(this.myClass('step')).
+        start('h3').add('Step 2: Analyze Structure & Select DAO').end().
+        start('div').addClass('mb-lg').
+          // Structure Analysis button
+          start('div').addClass('mt-lg').
+            start(this.ANALYZE_STRUCTURE).end().
+          end().
+          
+          // Structure Status and Results
+          start('div').addClass('mt-lg').
+            // Structure Analysis Results
+            show(this.filesVerified$).
+            start('div').addClass('p-sm').addClass(this.myClass('bg-light')).addClass('mb-lg').
+              start('h4').addClass('mb-sm').add('Structure Analysis Results').end().
+              start('div').addClass('mb-sm').addClass(this.myClass('text-success')).
+                add(this.uploadedFiles$.map(files => 
+                  `✅ Structure verified! All ${files ? files.length : 0} files have matching headers:`
+                )).
+              end().
+              start('div').addClass(this.myClass('text-mono')).addClass(this.myClass('bg-white')).addClass('p-sm').
+                add(this.detectedHeaders$.map(headers => headers.join(', '))).
+              end().
+            end().
+
+            // Target DAO Selection (only show after analysis)
+            show(this.filesVerified$).
+            start('div').addClass('mt-lg').
+              start('h4').add('Target DAO').end().
+              start('p').addClass(this.myClass('text-muted')).
+                add('Select the target DAO where the data will be uploaded.').
+              end().
+              start('div').addClass('mt-sm').
+                add(this.SELECTED_DAO).
+              end().
+            end().
```

Comment:
```suggestion
            tag(this.SELECTED_DAO.__)
```
Much simpler, same result, set the label on the property. Properties also have a supportingLabel property that does the same thing your text-muted class is doing

Reply:
@abdelaziz-mahdy same here